packages/base-crypto/src/VirgilCrypto.ts

```diff
@@ -156,18 +270,25 @@ export class VirgilCrypto implements ICrypto {
       recipientCipher.addKeyRecipient(identifier, lowLevelPublicKeys[index]);
     });
 
-    recipientCipher.startEncryption();
-    const messageInfo = recipientCipher.packMessageInfo();
-    const processEncryption = recipientCipher.processEncryption(myData);
-    const finishEncryption = recipientCipher.finishEncryption();
+    let result: BufferType | undefined;
+    try {
+      recipientCipher.startEncryption();
+      const messageInfo = recipientCipher.packMessageInfo();
+      const processEncryption = recipientCipher.processEncryption(myData);
+      const finishEncryption = recipientCipher.finishEncryption();
+      result = NodeBuffer.concat([messageInfo, processEncryption, finishEncryption]);
+    } catch (error) {
+      recipientCipher.delete();
+      aes256Gcm.delete();
+      lowLevelPublicKeys.forEach(lowLevelPublicKey => lowLevelPublicKey.delete());
+      throw error;
+    }
 
     recipientCipher.delete();
     aes256Gcm.delete();
-    lowLevelPublicKeys.forEach(lowLevelPublicKey => {
-      lowLevelPublicKey.delete();
-    });
+    lowLevelPublicKeys.forEach(lowLevelPublicKey => lowLevelPublicKey.delete());
 
-    return NodeBuffer.concat([messageInfo, processEncryption, finishEncryption]);
+    return result!;
```

Comment:
Return statement can be put inside the `try` block, then the `catch` block can be replaced with `finally` and there will be no need to duplicate the `.delete` invocations